doajtest/unit/test_bll_article_get_duplicates.py

```diff
@@ -119,3 +119,58 @@ def test_01_get_duplicates(self, name, kwargs):
             else:
                 assert duplicate is None
 
+    def test_02_get_duplicates_tricked(self):
+        account = Account()
+        account.add_role("publisher")
+        account.set_id("test")
+        account.save(blocking=True)
+
+        j = Journal()
+        jbj = j.bibjson()
+        jbj.add_identifier(jbj.E_ISSN, "1111-1111")
+        jbj.title = "Journal"
+        j.set_owner("test")
+        j.save(blocking=True)
+
+        a1 = Article()
+        bj1 = a1.bibjson()
+        bj1.add_identifier(bj1.E_ISSN, "1111-1111")
+        bj1.add_identifier(bj1.DOI, "10.1234/1111")
+        bj1.add_url("http://10.1234/1111", bj1.FULLTEXT)
+        bj1.title = "One"
+        a1.save(blocking=True)
+
+        a2 = Article()
+        bj2 = a2.bibjson()
+        bj2.add_identifier(bj2.E_ISSN, "1111-1111")
+        bj2.add_identifier(bj2.DOI, "10.1234/2222")
+        bj2.add_url("http://10.1234/2222", bj2.FULLTEXT)
+        bj2.title = "Two"
+        a2.save(blocking=True)
+
+        # this contains a mix of both ids
+        a3 = Article()
+        bj3 = a3.bibjson()
+        bj3.add_identifier(bj3.E_ISSN, "1111-1111")
+        bj3.add_identifier(bj3.DOI, "10.1234/1111")
+        bj3.add_url("http://10.1234/2222", bj3.FULLTEXT)
+        bj3.title = "Three"
+
+        # first check that what we see is two possible duplicates
+        dups = self.svc.get_duplicates(a3)
+        assert len(dups) == 2
+
+        # now do the update, which should overwrite one of the articles
+        # (we don't know which one)
+        self.svc.create_article(a3, account)
+
+        import time
+        time.sleep(2)
+
+        # check that either way round we look at it, we now have a duplicate
+        # in the system
+        dups = self.svc.get_duplicates(a1)
+        assert len(dups) == 1
+
+        dups = self.svc.get_duplicates(a2)
+        assert len(dups) == 1
```

Comment:
Tests fail on this line - `dups` is `[]`

Reply:
ah, probably it just needs a rebase

Reply:
I can look at it if you like, I'll try to get to it this afternoon

Reply:
Ok, this should be fixed.  The test was used to demonstrate an error, and by passing showed the problem.  Now that it is failing, the problem is fixed, and the checks were already incorporated into the main test, so I have just removed the experimental test.